oio/xcute/orchestrator.py

```diff
@@ -245,32 +248,29 @@ def dispatch_tasks(self, beanstalkd_workers,
             raise ValueError('Task payload is too big (length=%s)' %
                              len(beanstalkd_payload))
 
+        sent = False
         while self.running:
-            workers_tried = set()
-            for worker in beanstalkd_workers:
-                if worker is None:
-                    self.logger.info('No beanstalkd available (job_id=%s)',
-                                     job_id)
-                    break
-
-                if worker.addr in workers_tried:
-                    self.logger.debug('Tried all beanstalkd (job_id=%s)',
-                                      job_id)
+            for beanstalkd_worker in beanstalkd_workers:
+                if not self.running:
+                    return sent
+                if beanstalkd_worker is not None:
                     break
 
-                sent = worker.send_job(beanstalkd_payload)
-
-                if not sent:
-                    workers_tried.add(worker.addr)
-
-                    continue
+            try:
+                beanstalkd_worker.put(beanstalkd_payload)
+                sent = True
+                self.logger.debug(
+                    '[job_id=%s] Tasks sent to %s: %s', job_id,
+                    beanstalkd_worker.addr, str(tasks))
+                break
+            except Exception as exc:
+                self.logger.warn(
+                    '[job_id=%s] Fail to send beanstalkd job: %s',
+                    job_id, exc)
+                beanstalkd_worker.is_broken = True
```

Comment:
Maybe be a little more lenient here, just like it's usually done for heartbeats: count the number of failed requests in a row, and if it's over a certain threshold, consider it as broken.

Reply:
If there was an error, it is surely because of a problem. So let's leave it alone until the next refreshment which is coming soon (30 seconds).

Reply:
A small error can happen anytime, it doesn't mean the beanstalkd is dead or dying. Maybe it's good enough for now, but I'd add a comment just saying that we could be more lenient and wait for a few errors in a row to happen before marking it as broken.